### ISSUE TITLE
Filter problem with boolean "Field true not found on type..."

### ISSUE THREAD

**Issue by atorres16** [EntityGraphQL/EntityGraphQL#314]
I'm following the docs 
https://entitygraphql.github.io/docs/field-extensions/filtering/

I don't know what I'm doing wrong but this query

```graphql
{
  sites(filter: "isActive == true") {
    id
    name
    isActive
  }
}

```

Fails with: 
```graphql
{
  "errors": [
    {
      "message": "Field true not found on type Site"
    }
  ]
}
```

There's something wrong only with booleans because these two work fine:

```graphql
{
  sites(filter: "id == 1") {
    id
    name
    isActive
  }
}


{
  sites(filter: "name == \"My Site\"") {
    id
    name
    isActive
  }
}
```

I'm decorating my DbSets with [UseFilter]

```csharp
   [UseFilter]
   public DbSet<Site>? Sites { get; set; }
```

This is my Site model

```csharp
 public class Site 
    {
     
        [Key]
        public int Id { get; set; }

        [Required]
        [MaxLength(150)]
        public string Name { get; set; }

        public bool IsActive { get; set; } = true;
```

Program.cs

```csharp
builder.Services.AddDbContext<Db>(ip => ip.UseSqlServer(builder.Configuration.GetConnectionString("Db")));
builder.Services.AddGraphQLSchema<Db>();

app.UseEndpoints(routeBuilder =>
{
    routeBuilder.MapControllers();
    routeBuilder.MapGraphQL<Db>();
    routeBuilder.MapGraphQLAltair(new AltairOptions
    {
        GraphQLEndPoint = PathString.FromUriComponent("/graphql")
    });
});
```

I'm using:
* EntityGraphQL.AspNet 5.0.0
* .NET 6
* EF Core 6.0.21

**Comment by atorres16** [comment]
It fails when it tries to build the right side of the expression

EntityGraphQL.Compiler.EntityQuery.EntityQueryNodeVisitor:

```csharp
  public override Expression VisitBinary(EntityQLParser.BinaryContext context)
        {
            var left = Visit(context.left);
            var right = Visit(context.right);
            var op = MakeOperator(context.op.GetText());
```

There must be something wrong with that antlr4 thing, that's as far I could get

**Comment by atorres16** [comment]
It has something to do with the way the string is parsed, because this works:

```
{
  people(filter:"isDeleted == (1==2))"){
    firstName
  }
}
```

**Comment by atorres16** [comment]
I don't know anything about ANTLR or parsers, but if I delete true and false from this part of EntityQL.g4 then it works
```

// identity includes keywords too
identity: ID
	| 'true' DELETE THIS
	| 'false' DELETE THIS
	| 'and'
	| 'if'
	| 'then'
	| 'else';
```

**Comment by lukemurray** [comment]
Thanks for the input @kaby76!

@atorres16 thanks for the details too. I should have a change to look at this over the next few days.

**Comment by atorres16** [comment]
@lukemurray thanks, again I'm not an expert, but I can see "true" and "false" duplicated in the g4 file

```
....
// identity includes keywords too
identity: ID
	| 'true'    <------ HERE
	| 'false'
	| 'and'
	| 'if'
	| 'then'
	| 'else';

int: '-'? DIGIT+;
decimal: '-'? DIGIT+ '.' DIGIT+;
boolean: 'true' | 'false'; <----- HERE

```

maybe that's causing the conflict? idk, but I wanted to highlight that.

**Comment by atorres16** [comment]
@kaby76 sorry about the extra parenthesis, it was a typo, I just wanted to say that (1==1) resolves to true and (1==2) resolves to false.

**Comment by lukemurray** [comment]
@atorres16 Note this is fixed in 5.0.1. The duplication you noted was a hang over from long ago and caused the parse of "true" is ambiguous issue described above.

Let me know if you have other issues. There are some further clean ups to the grammar etc coming.

**Comment by kaby76** [comment]
> @kaby76 sorry about the extra parenthesis, it was a typo, I just wanted to say that (1==1) and (1==2) return true/false

The extra parenthesis mistake was actually a blessing. That is because the start rule must be an EOF-terminated rule. As I point out, the grammar actually accepted the bad input without producing any error. It's another thing that I wish Antlr would do differently, but apparently that is what the designers want.
